Exscript/protocols/ssh2.py

```diff
@@ -269,6 +269,8 @@ def _paramiko_auth_autokey(self, username, password):
 
     def _get_auth_methods(self, allowed_types):
         auth_methods = []
+        if self.auth_methods:
```

Comment:
Wouldn't it make more sense to do something like

```
auth_method_handlers = []
if self.auth_methods:
    auth_methods = [m for m in self.auth_methods if m in allowed_types]
else:
    auth_methods = allowed_types
for method in auth_methods:
    for type_name in auth_types[method]:
        auth_method_handlers.append(getattr(self, type_name))
return auth_method_handlers
```

Otherwise Exscript wouldn try to authenticate using unsupported methods.

Reply:
You are absolutely right. Thanks for the review - I will change the code.